components/esp_lvgl_port/include/esp_lvgl_port.h

```diff
@@ -61,6 +62,7 @@ typedef struct {
     esp_lcd_panel_handle_t panel_handle;    /*!< LCD panel handle */
     uint32_t    buffer_size;    /*!< Size of the buffer for the screen in pixels */
     bool        double_buffer;  /*!< True, if should be allocated two buffers */
+    uint32_t    trans_size;     /*!< Allocated buffer will be in SRAM to move framebuf */
```

Comment:
I think, better name for it is: `sw_rotation_buffer` or somethink like that.

Reply:
>Why is it named trans_size, not sw_rotation_buffer?

- The operation of sw_rotate is inside LVGL, and I originally planned to do sw_rotate in lvgl_port_flush_callback, but considering that LVGL supports sw_rotation inside, so I've deleted this part.

- The function of this buffer here is that when the frame_buffer is in the PSRAM, it provides a small buffer of internal SRAM to deal with it, so SPI driver no needs to do malloc and free for each transfer internally. It is user-configurable and more flexible.

>I tested in on HW, but I don't see any difference in all cases.

The phenomenon is like the screenshot I gave you, there will be an oblique tear, which is more obvious on the customer's product. 
Let me think how to show it. I tried it on box, but I can't show it well.

Reply:
When I'm testing the demo, I found some compatibility issues in lvgl 8.3.9 and I'm confirming.

Reply:
This is a 240*320 resolution screen, and customers need to rotate 90 degrees to use it horizontally.
[LCD.png](https://cowtransfer.com/s/c2b2b50664d940)

When using hardware rotation, a more noticeable oblique tear can be seen.
[ hw_roate.mp4 ](https://cowtransfer.com/s/185c869d77704b) 
[ sw_rotate.mp4 ](https://cowtransfer.com/s/4b3b65af8df848)

This is a comparison of the boot video, using full-screen framebuf + 10-line SRAM handling and 10-line SRAM rendering, and the tearing of the interface is also significantly improved.
[ poweron.mp4 ](https://cowtransfer.com/s/7cad37fcd7e748)

Reply:
> better name for it is: sw_rotation_buffer or something like that

i'd also like to see more descriptive name for this struct member